docs/concept/clustering.md

```diff
@@ -0,0 +1,158 @@
+
+- Meta Nodes share high-level metadata about the cluster.
+- Data Nodes store and manage the raw time series data and communicate with Meta Nodes.
+- Query Nodes interact with Data Nodes to execute queries and return results to the Liaison Nodes.
+- Liaison Nodes distribute incoming requests to the appropriate Query Nodes or Data Nodes.
+
+## 3. **Data Organization**
+
+Different nodes in BanyanDB are responsible for different parts of the database, while Query and Liaison Nodes manage the routing and processing of queries.
+
+### 3.1 Meta Nodes
+
+Meta Nodes store all high-level metadata that describes the cluster. This data is kept in an etcd-backed database on disk, including information about the shard allocation of each Data Node. This information is used by the Liaison Nodes to route data to the appropriate Data Nodes, based on the sharding key of the data.
+
+By storing shard allocation information, Meta Nodes help ensure that data is routed efficiently and accurately across the cluster. This information is constantly updated as the cluster changes, allowing for dynamic allocation of resources and efficient use of available capacity.
+
+### 3.2 Data Nodes
+
+Data Nodes store all raw time series data, metadata, and indexed data. On disk, the data is organized by `<group>/shard-<shard_id>/<segment_id>/`. The segment is designed to support retention policy.
+
+### 3.3 Query Nodes
+
+Query Nodes do not store data. They handle the computational tasks associated with data queries, interacting directly with Data Nodes to execute queries and return results.
+
+### 3.4 Liaison Nodes
+
+Liaison Nodes do not store data but manage the routing of incoming requests to the appropriate Query or Data Nodes. They also provide authentication, TTL, and other security services.
+
+## 4. **Determining Optimal Node Counts**
+
+When creating a BanyanDB cluster, choosing the appropriate number of each node type to configure and connect is crucial. The number of Meta Nodes should always be odd, for instance, “3”. The number of Data Nodes scales based on your storage and query needs. The number of Query Nodes and Liaison Nodes depends on the expected query load and routing complexity.
+
+The BanyanDB architecture allows for efficient clustering, scaling, and high availability, making it a robust choice for time series data management.
+
+## 5. Writes in a Cluster
+
+In BanyanDB, writing data in a cluster is designed to take advantage of the robust capabilities of underlying storage systems, such as Google Compute Persistent Disk or Amazon S3(TBD). These platforms ensure high levels of data durability, making them an optimal choice for storing raw time series data.
+
+Unlike some other systems, BanyanDB does not support application-level replication. Instead, it delegates the task of replication to these underlying storage systems. This approach simplifies the BanyanDB architecture and reduces the complexity of managing replication at the application level.
+
```

Comment:
Besides reducing complexity, I think another point is that for the Big data system, this way can also significantly reduce the cost of storing a large amount of data, which I think is very beneficial to enterprise production.